storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/RocksDbDatabase.java

```diff
@@ -65,6 +69,18 @@ public static Database createV3(
     return createV3(metricsSystem, db, stateStorageMode);
   }
 
+  public static Database createV4(
+      final MetricsSystem metricsSystem,
+      final RocksDbConfiguration hotConfiguration,
+      final RocksDbConfiguration finalizedConfiguration,
+      final StateStorageMode stateStorageMode) {
+    final RocksDbAccessor hotDb =
+        RocksDbInstanceFactory.create(hotConfiguration, V4SchemaHot.class);
+    final RocksDbAccessor finalizedDb =
+        RocksDbInstanceFactory.create(finalizedConfiguration, V4SchemaFinalized.class);
```

Comment:
Probably not for this PR, but I wonder if when storage mode is `PRUNE` we should skip loading or creating the archive db entirely and just pass in a `StubRocksDbAccessor` here.  Then it would automatically discard archive state and we avoid the overhead of opening a second RocksDB instance that we never actually use.

Reply:
#2197